### DISCUSSION TITLE
Offcanvas execute multiple time

### DISCUSSION THREAD

**Discussion by krosoftware** [twbs/bootstrap#36942] [General] [answered]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

I have simple Bootstrap 5.2 Offcanvas example here: https://stackblitz.com/edit/qw6rdj?file=index.html

Please try to click on button more than once. After each click, you will see +1 alert on open and close.

Why is that?

How to display alert (open and close message) only once on each click? Thank you

### Reduced test cases

https://stackblitz.com/edit/qw6rdj?file=index.html

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome, Microsoft Edge, Opera

### What version of Bootstrap are you using?

v5.2

**Comment by monsterRocky** [comment]
Compress with RTL and join along with css.

**Comment by GeoSot** [comment, marked as answer]
Just try to register your listeners once:

```js
element.body.addEventListener('myEvent', () => {
    console.log('I run only once! 😇');
  }, { once: true });
```

**Reply by krosoftware** [reply]
Thanks but Event should be fired only once by default, I think.
